### ISSUE TITLE
Check if scikit-learn has functions that we can reuse

### ISSUE THREAD

**Issue by haesleinhuepf** [BiAPoL/biapol-utilities#3]
Hi Johannes @jo-mueller,

could you please check if this implementation of the Jaccard score can be of use for us?
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.jaccard_score.html

Does it do the right thing, e.g. for comparing label images? Is it performing well?

Thanks!

Best,
Robert

**Comment by jo-mueller** [comment]
Started to implement sklearn functions [here](https://github.com/BiAPoL/biapol-utilities/tree/compare-labels)

**Comment by haesleinhuepf** [comment]
Hi Johannes @jo-mueller ,

sorry for phrasing my idea a bit unclearly: I was hoping that we can replace some the `biapol-utilities` code with `scikit-learn` code. I was hoping that their implementation is faster. As I obviously pointed to the wrong method, I digged a bit deeper. 

I'm now suggesting to replace `biapol_utilities.label_overlap_matrix` with [`sklearn.metrics.confusion_matrix`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html). As far as I can see, they do the same. In case of large images, the `scikit-learn` version is much faster. At label image sizes of > 40 MB, our method is hardly benchmarkable anymore. But typical lightsheet data is even larger. 

I uploaded a quick and dirty benchmaring notebook here:

https://github.com/BiAPoL/biapol-utilities/blob/compare_overlap_matrices/notebooks/label/overlap_mattrix_benchmarking.ipynb

Let me know what you think!

Thanks!

Best,
Robert

**Comment by jo-mueller** [comment]
I implemented the confusion matrix in `compare_overlap_matrices` and kept the old function as a legacy function (now called `label_overlap_matrix_numpy`. 

While testing I found a major/minor (point of discussion) weakness of the current matching strategy: The script runs into trouble if labels are scarce, e.g. `[0, 1, 1, 5, 5, 6, 6]`, or to be more precise, if there are long stretches of unoccupied labels in the images. If the labels in the two input image are sequential, then this is not a problem.

**Comment by haesleinhuepf** [comment]
> and kept the old function as a legacy function (now called `label_overlap_matrix_numpy`.

I suggest removing that _legacy_ function given the age of it.

> If the labels in the two input image are sequential, then this is not a problem.

Can we relabel the label images internally? Or are you suggesting that properly labeled images are a precondition that should be mentioned in the documentation?

Looking forward to a pull request!

**Comment by jo-mueller** [comment]
> Looking forward to a pull request!

Here goes - I added it to the implementation of the speed benchmark, which now shows comparable performance of sklearn and biapol. (PR #20)

> I suggest removing that legacy function given the age of it.

Also part of the PR.

> Can we relabel the label images internally? Or are you suggesting that properly labeled images are a precondition that should be mentioned in the documentation?

I *partially* added this to the PR in ```match_labels(label_image_x, label_image_y, ...)```:
```python
label_image_y, _, _ = relabel_sequential(label_image_y)  # relabel only label_image_y
```

Note that only relabel the 2nd input, because it's labels will be changed anyway. I'm not so certain about relabeling `label_image_x`. Although `relabel_sequential` from skimage does provide a lookup table to invert the relabeling after matching the labels, it can only allow to relabel *known labels*.  

TL; DR: Relabeling label_image_x and inverting this process afterwards may lead to problems and I'm not exactly sure when or how exactly this would behave so I'd refrain from doing it.
